app/classes/amity.py

```diff
@@ -218,61 +220,80 @@ def print_allocations(self, filename="None"):
 		""" A function that prints rooms and its occupants"""
 
 		# Write allocated rooms and its occupants into a file
-		if filename != "None":
-			file = open(filename, 'w')
-			record = set()			
-			for room in self.allocated_rooms:
-				if room.name not in record:
-					record.add(room)
-			for sortedrooms in record:
-				file.write('\n' + "%s\n" % sortedrooms.name + '-'*50 + '\n')
-				for occupant in sortedrooms.occupants:
-					if len(sortedrooms.occupants)>0:
-						file.write("%s\n" % occupant.name )
```

Comment:
Is there a way we can avoid calling the write function to the same file twice? That would be nice. Take a scenario where you sending sms and first you send the title then the body, if we can send the whole text at once would be more efficient.